app/models/miq_alert.rb

```diff
@@ -72,6 +73,14 @@ def set_responds_to_events
     self.responds_to_events = events unless events.nil?
   end
 
+  def validate_delay_next_evaluation
+    next_frequency = (options || {}).fetch_path(:notifications, :delay_next_evaluation)
+    evaluation_method = expression[:eval_method] if expression.kind_of?(Hash)
+    if !evaluation_method.nil? && evaluation_method == "dwh_generic" && !next_frequency.nil? && next_frequency != 0
```

Comment:
@moolitayer Instead of hardcoding `dwh_generic` here, how about adding a flag to the definition of the expression indicating that is should always be evaluated. Something like this -
```diff
diff --git a/app/models/miq_alert.rb b/app/models/miq_alert.rb
index f4db271..0027abf 100644
--- a/app/models/miq_alert.rb
+++ b/app/models/miq_alert.rb
@@ -442,7 +442,7 @@ class MiqAlert < ApplicationRecord
           {:name => :value_mw_garbage_collector, :description => _("Duration Per Minute (ms)"), :numeric => true}
         ]},
       {:name => "dwh_generic", :description => _("All Datawarehouse alerts"), :db => ["ContainerNode"], :responds_to_events => "datawarehouse_alert",
-        :options => []}
+        :options => [], :always_evaluate => true}
     ]
   end
```
Then you can check that here with something like `expression_by_name('dwh_generic')[:always_evaluate]`. The same can be done on the UI side. What do you think?